lib/git-hooks.js

```diff
@@ -33,8 +33,26 @@ var HOOKS = [
 function excludeIgnoredPaths(hooks, callback) {
     exec('git check-ignore ' + hooks.join(' '), function (error, output) {
         // intentionally ignore errors
-        callback(hooks.filter(function (hookName) {
-            return output.indexOf(hookName) === -1;
+        output = typeof output !== 'string' ? '' : output; // basic sanity check
+        var ignoredHookPaths = output.split(/\r?\n/); // split string into array of each ignored file
+
+        ignoredHookPaths = ignoredHookPaths.filter(function (ignoredHookPath) {
+            // remove empty strings from the results
+            return ignoredHookPath.length > 0;
+        });
+        var normalizePath = function (hookPath) {
+            // normalize paths
+            if (process.platform === 'win32') {
```

Comment:
Let's remove windows support in this pr. It's better to add it in a separate pr.